### PR TITLE
Add Aws CI/CD recipe

### PR DESCRIPTION
**Pull request by jeanluciano**
## Description

What's cooking? 
If this PR references an issue, please attach to the existing issue.

## Type of change

- [ x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## New Recipe Checklist

- [ x] My PR is in the format of `Add <project-name> recipe`
- [x ] My recipe is reproducible and explains everything needed to run successfully. If my code has external dependencies, I make mention of them.
- [ x] My code is easily understandable and/or well-commented.
- [x ] If my recipe requires a new category (e.g. creating a monitoring/ folder in devops/), I have encapsulated my project within its own subfolder so that others can add their own recipes as well.
- [x ] If my recipe uses Prefect < 2.0, my code is within the prefect-v1-legacy/ folder.

### CODE REVIEW

**Review by serinamarie — APPROVED** [review]

**Review by robfreedy — APPROVED** [review]

File: devops/github-actions/s3-storage-cicd/.github/workflows/dev.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    steps:
+      - uses: actions/checkout@master
+      - uses: jakejarvis/s3-sync-action@master
+        with:
+          args: --exclude '.venv/*' --exclude '.git/*'
+        env:
+          AWS_S3_BUCKET: ${{ secrets.AWS_S3_BUCKET }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          DEST_DIR: "/dev"
```
Comment (zzstoatzz) [inline devops/github-actions/s3-storage-cicd/.github/workflows/dev.yml:20]: I don't understand why we're syncing a folder to s3, isn't this stepping on the toes of the `prefect deployment build` command?

**Review by zzstoatzz — CHANGES_REQUESTED** [review]

File: devops/github-actions/s3-storage-cicd/.github/workflows/dev.yml (hunk 1 above)
Comment (serinamarie) [reply devops/github-actions/s3-storage-cicd/.github/workflows/dev.yml:20]: That's a good point. I would have expected a repo checkout step, then a `prefect deployment build` step.

**Review by serinamarie — CHANGES_REQUESTED** [review]

**Comment by jeanluciano** [comment]
@zzstoatzz @serinamarie If we run deployments build, it will recreate the deployment file. This lets us create a deployment once and then sync our code data by pushing to repo without having to rebuild a deployment every time.

**Comment by gabcoyne** [comment]
I don't think this is an approach that we should advise, as of most recent releases deployments will respect previous configuration in both YAML and in the API, so regenerating deployments has no downside

**Pull request closed without merging**